tg/tg_module.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2024 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+package tg
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"slices"
+	"sort"
+
+	"github.com/gruntwork-io/go-commons/env"
+	"github.com/gruntwork-io/terragrunt/config"
+	"github.com/gruntwork-io/terragrunt/configstack"
+	"github.com/gruntwork-io/terragrunt/options"
+	"github.com/gruntwork-io/terragrunt/util"
+	"github.com/rs/zerolog/log"
+	"github.com/terramate-io/terramate/errors"
+	"github.com/terramate-io/terramate/project"
+	"github.com/zclconf/go-cty/cty/function"
+)
+
+// ErrParsing indicates there is an error parsing a Terragrunt file.
+const ErrParsing errors.Kind = "parsing Terragrunt file"
+
+type (
+	// Module is a Terragrunt module.
+	Module struct {
+		Path       project.Path  `json:"path"`
+		ConfigFile project.Path  `json:"config"`
+		DependsOn  project.Paths `json:"depends_on"`
+	}
+
+	// Modules is a list of Module.
+	Modules []*Module
+)
+
+// ScanModules scans dir looking for Terragrunt modules. It returns a list of
+// modules with its "trigger paths" computed. The TriggerPaths are paths that,
```

Comment:
Where are these trigger paths? Is this just Path?

Reply:
I changed the name to `DependsOn` and forgot to update the docs. 😬 
thanks for catching